site/releases/2024-jan-22/highlights.qmd

```diff
@@ -0,0 +1,218 @@
+
+### ValidMind Developer Framework (v1.25.3)
+
+<!---[SC-2628] `init_model` improvements  by @AnilSorathiya in [#304](https://github.com/validmind/developer-framework/pull/304)--->
+
+#### Improvements to `init_model`
+
+When initializing a model, you can now pass a dataset with pre-computed model predictions to improve performance. By default, if no prediction column is specified when calling `init_model`, the ValidMind Developer Framework will compute the model predictions as before.
+
+To illustrate how passing a dataset that includes a prediction column can help, consider the following example without a prediction column: 
+
+```python
+vm_model = vm.init_model(
+    model,
+    train_ds=vm_train_ds,
+    test_ds=vm_test_ds,
+)
+```
+
+Internally, this example invokes the `predict()` method of the model for the training and test datasets when the model is initialized. This approach can be problematic with large datasets: `init_model` can simply take too long to compute.
+
+You can now avoid this issue by providing a dataset with a column containing pre-computed predictions, similar to the following example. If `init_model` detects this column, it will not generate model predictions at runtime. 
+
+|x1|x2|...|target_column|prediction_column|
+|-|-|-|-|-|
+|0.1|0.2|...|0|0|
+|0.2|0.4|1...|1|1|
+
+Usage example with a prediction column:
+
+```python
+vm.init_dataset(
+     dataset=df,
+     feature_columns=[...],
+     target_column= ...,
+     extra_columns={
+         prediction_column: 'NAME-OF-PREDICTION-COLUMN',
+    },
+)
+```
+
+<!---[SC-2438] Support for adding feature columns parameter to init dataset by @juanmleng in [#279](https://github.com/validmind/developer-framework/pull/279)--->
+#### Improvements to `init_dataset`
+
+When initializing a dataset, the new `feature_columns` argument lets you specify a list of feature names for prediction to improve efficiency. Internally, the function filters the dataset to retain only these specified features for prediction-related tasks, leaving the remaining dataset available for other purposes, such as segmentation. 
+
+This improvement replaces the existing behavior of `init_dataset`, which loaded the entire dataset, incorporating all available features for prediction tasks. While this approach worked well, it could impose limitations when generating segmented metrics and proved somewhat inefficient with large datasets containing numerous features, of which only a small subset were relevant for prediction.
+
+Usage example:
+
+```python
+feature_columns = ['CreditScore', 'Age', 'Balance', 'NumOfProducts', 'EstimatedSalary']
+
+vm_train_ds = vm.init_dataset(
+    dataset=train_df,
+    target_column=demo_dataset.target_column,
+    feature_columns=feature_columns
+)
+```
+
+Additionally, two new metrics enable you to report by segment and show the use of `feature_columns`:
```

Comment:
Any metric that uses a dataset is affected by custom feature columns :).

```suggestion
```

Reply:
I removed this manually as we also had a couple of actual metrics to remove right below.